source/slang/glsl.meta.slang

```diff
@@ -417,3 +252,2047 @@ ${{{{
+public typealias sampler2DArray = Sampler2DArray<float4>;
+
+public typealias usamplerCubeArray = SamplerCubeArray<uint4>;
+public typealias isamplerCubeArray = SamplerCubeArray<int4>;
+public typealias samplerCubeArray = SamplerCubeArray<float4>;
+
+__generic<let sampleCount:int=0, let format:int=0>
+public typealias sampler1DArrayShadow = __TextureImpl<
+    float,
+    __Shape1D,
+    1, // isArray
+    0, // isMS
+    sampleCount,
+    0, // access
+    1, // isShadow
+    1, // isCombined
+    format,
+    0  // isRectangle
+>;
+
+__generic<let sampleCount:int=0, let format:int=0>
+public typealias sampler2DArrayShadow = __TextureImpl<
+    float,
+    __Shape2D,
+    1, // isArray
+    0, // isMS
+    sampleCount,
+    0, // access
+    1, // isShadow
+    1, // isCombined
+    format,
+    0  // isRectangle
+>;
+
+__generic<let sampleCount:int=0, let format:int=0>
+public typealias samplerCubeArrayShadow = __TextureImpl<
+    float,
+    __ShapeCube,
+    1, // isArray
+    0, // isMS
+    sampleCount,
+    0, // access
+    1, // isShadow
+    1, // isCombined
+    format,
+    0  // isRectangle
+>;
+
+public typealias sampler2DMS = Sampler2DMS<float4>;
+public typealias isampler2DMS = Sampler2DMS<int4>;
+public typealias usampler2DMS = Sampler2DMS<uint4>;
+
+__generic<T=float4, let sampleCount:int=0, let format:int=0>
+public typealias Sampler2DMSArray = Sampler2DArrayMS<T, sampleCount, format>;
+public typealias sampler2DMSArray = Sampler2DMSArray<float4>;
+public typealias isampler2DMSArray = Sampler2DMSArray<int4>;
+public typealias usampler2DMSArray = Sampler2DMSArray<uint4>;
+
+__generic<T=float4, let sampleCount:int=0, let format:int=0>
+public typealias Sampler2DRect = __TextureImpl<
+    T,
+    __Shape2D,
+    0, // isArray
+    0, // isMS
+    sampleCount,
+    0, // access
+    0, // isShadow
+    1, // isCombined
+    format,
+    1  // isRectangle
+>;
+
+public typealias sampler2DRect = Sampler2DRect<float4>;
+public typealias isampler2DRect = Sampler2DRect<int4>;
+public typealias usampler2DRect = Sampler2DRect<uint4>;
+
+public typealias sampler2DRectShadow = __TextureImpl<
+    float,
+    __Shape2D,
+    0, // isArray
+    0, // isMS
+    0, // sampleCount
+    0, // access
+    1, // isShadow
+    1, // isCombined
+    0, // format
+    1  // isRectangle
+>;
+
+//#define TEST_samplerBuffer
+#if defined(TEST_samplerBuffer)
+
+// TODO: We need to properly implement Shader Storage Buffer Object, or SSBO for short.
+__generic<T, let format:int=0>
+public typealias SamplerBuffer = __TextureImpl<T, __ShapeBuffer, 0, 0, 0, 1 /*RW*/, 0, 0, format>;
+public typealias samplerBuffer = SamplerBuffer<vec4>;
+public typealias isamplerBuffer = SamplerBuffer<int4>;
+public typealias usamplerBuffer = SamplerBuffer<uint4>;
+
+#endif // #if defined(TEST_samplerBuffer)
+
+
+// -------------------
+// textureSize
+// -------------------
+
+__generic<T:__BuiltinArithmeticType, let N:int>
+__target_intrinsic(glsl)
```

Comment:
Going forwarrd we are going to remove `__target_intrinsic` and have everything defined in `__target_switch`. 
The `__target_intrinsic` here is not necessary so let's remove them here.

glsl.meta.slang should only make use of __target_switch.

Reply:
I was wondering about why none of functions in glsl.meta.slang didn't have it. Your explanation makes sense. I will remove it from all functions in glsl.meta.slang